### PR TITLE
[PM-10024] Force focus on Master Password or Pin input field.

### PR DESCRIPTION
**Pull request by qwexter**
## 🎟️ Tracking

[issues/3383](https://github.com/bitwarden/android/issues/3383)

## 📔 Objective

Request autofocus on input field to show keyboard on Unlock Vault screen if biometric unlock is disabled or unavailable.
Also extent input field to support keyboard actions and and bind Unlick action to keyboard Done event too.
Add tests to cover cases with empty PIN\Password and ingnore Unlock action  and focus related tests too.

## Screenshots \ Video

demo_password

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockScreen.kt [hunk 1]
```diff
@@ -196,6 +198,13 @@ fun VaultUnlockScreen(
                             .testTag(state.vaultUnlockType.unlockScreenInputTestTag)
                             .padding(horizontal = 16.dp)
                             .fillMaxWidth(),
+                        autoFocus = state.showKeyboard,
+                        imeAction = ImeAction.Done,
+                        keyboardActions = KeyboardActions(
+                            onDone = remember(viewModel) {
+                                { viewModel.trySendAction(VaultUnlockAction.UnlockClick) }
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockScreen.kt:205]: Can we add a test for this

Comment (qwexter) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockScreen.kt:205]: done

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModel.kt [hunk 2]
```diff
@@ -208,6 +208,9 @@ class VaultUnlockViewModel @Inject constructor(
 
     private fun handleUnlockClick() {
         val activeUserId = authRepository.activeUserId ?: return
+
+        if (state.input.isEmpty()) return
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModel.kt:212]: Can this ever happen?
We already disable the button when the text is blank.

```kotlin
        isEnabled = state.input.isNotEmpty(),
```

Comment (SaintPatrck) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModel.kt:212]: With the introduction of KeyboardAction handling this can occur if Done is pressed on the keyboard when input is empty. 

Comparing behavior with the MAUI app, an error dialog should be displayed stating "The Master password field is required." This change in behavior needs to be approved by our product team, or the existing behavior needs to be modified so that it matches MAUI app behavior. 

@qwexter please extract KeyboardAction related changes to a separate PR so that they can be considered independently of the changes related to input fields taking focus.

**Review by SaintPatrck — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModel.kt (hunk 2 above)
Comment (qwexter) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModel.kt]: I move out this behavior from PR.

**Review by david-livefront — APPROVED** [review]
